File: chapter_2/min_5_in_array.py
def five_small(list1):
    five = list1[:5]
    rest = list1[5:]

    for i in rest:
        m = max(five)
        if i < m: 
            five.remove(m)
            five.append(i)

    return five

if __name__ == '__main__':
    a = [112, 32, 344 ,45 , 7, 22, 78, 40, 56, 11]
    print(five_small(a))

File: min_5_in_array.rs
fn five_small(array: &[i32]) -> Vec<i32> {
    let mut five= Vec::from(&array[0..5]);
    let rest = &array[5..];

    rest.iter().fold(&mut five, |acc, x| {
        let m = acc.iter().max().unwrap();
        if x < m {
            let mindex = acc.iter().position(|v|*v == *m).unwrap();
            acc.remove(mindex);
            acc.push(*x);
            return acc;
        } else {
            acc
        }
    });
    five
}

fn main() {
    let a = &[112, 32, 344 ,45 , 7, 22, 78, 40, 56, 11];
    println!("{:?}", five_small(a));
}